src/legacy/ui/public/agg_types/agg_params.ts

```diff
@@ -71,11 +68,16 @@ function AggParams(params) {
  *         output object which is used to create the agg dsl for the search request. All other properties
  *         are dependent on the AggParam#write methods which should be studied for each AggType.
  */
-AggParams.prototype.write = function (aggConfig, aggs, locals) {
-  const output = { params: {} };
+export const writeParams = (
+  params: AggParam[],
+  aggConfig: AggConfig,
+  aggs?: AggConfigs,
+  locals?: Record<string, any>
+) => {
+  const output = { params: {} as Record<string, any> };
```

Comment:
Is `any` necessary or would `Record<string, AggParam>` work?

Reply:
no, this is the output, not the AggParam, and could be anything